pkg/client/query.go

```diff
@@ -116,19 +116,16 @@ func (c *QueryClient) run() {
 
 	ticker := time.NewTicker(c.cfg.QueryInterval)
 
-	for {
-		select {
-		case <-ticker.C:
-			c.runQueries()
-		}
+	for range ticker.C {
+		c.runQueries()
 	}
 }
 
 func (c *QueryClient) runQueries() {
 	// Compute the query start/end time.
 	start, end, ok := c.getQueryTimeRange(time.Now().UTC())
 	if !ok {
-		level.Debug(c.logger).Log("msg", "skipped querying because no eligible time range to query")
+		level.Debug(c.logger).Log("msg", "skipped querying because no eligible time range to query") //nolint:errcheck
```

Comment:
[nit] Could be worth having a global ignore for this.

Reply:
Do you mean a global ignore for the `errcheck` linter? Or an ignore for errors returned by `Log()`?

Reply:
> Or an ignore for errors returned by Log()?

This one. We already do it in Mimir (you can check the config there).

Reply:
Done, see #7.